qutip/tests/core/test_gates.py

```diff
@@ -142,3 +150,49 @@ def test_gate_normalises_pauli_group(self, gate):
                     del pauli_gates[i]
                     break
         assert len(pauli_gates) == 0
+
+
+@pytest.mark.parametrize("dtype", [qutip.data.Dense, qutip.data.CSR])
```

Comment:
I choose 2 types so we don't only get the default, but not all type since we are not testing the conversion code, just that it is used. Adding `dia` would just add more test without testing anything new.